src/components/query-viewer/query-viewer.tsx

```diff
@@ -0,0 +1,332 @@
+  /**
+   * Ax to show data labels
+   */
+  @Prop() showDataLabelsIn: string;
+  /**
+   * Timeline
+   */
+  @Prop() plotSeries: string;
+  /**
+   * Labels for XAxis
+   */
+  @Prop() xAxisLabels: string;
+  /**
+   * if true the x Axes intersect at zero
+   */
+  @Prop() xAxisIntersectionAtZero: boolean;
+  /**
+   * if true show values on the graph
+   */
+  @Prop() showValues: boolean;
+  /**
+   * X Axis title
+   */
+  @Prop() xAxisTitle: string;
+  /**
+   * Y Axis title
+   */
+  @Prop() yAxisTitle: string;
+  /**
+   * Type of data to show
+   */
+  @Prop() showDataAs: string;
+  /**
+   * If true includes trend on the graph
+   */
+  @Prop() includeTrend: boolean;
+  /**
+   * If includeTrend, defines the period of the trend
+   */
+  @Prop() trendPeriod: string;
+  /**
+   * For timeline for remembering layout
+   */
+  @Prop() rememberLayout: boolean;
+  /**
+   * Orientation of the graph
+   */
+  @Prop() orientation: string;
+  /**
+   * Include spark line
+   */
+  @Prop() includeSparkline: boolean;
+  /**
+   * Include max and min
+   */
+  @Prop() includeMaxMin: boolean;
+  /**
+   * Theme for showing the graph
+   */
+  @Prop() theme: string;
+  /**
+   * Object type -> Query or DataProvider
+   */
+  @Prop() objectType: string;
+  /**
+   * True if it is external query
+   */
+  @Prop() isExternalQuery: boolean;
+  /**
+   * Allowing elements order to change
+   */
+  @Prop() allowElementsOrderChange: boolean;
+  /**
+   * If type== PivotTable or Table, if true will shrink the table
+   */
+  @Prop() autoResize: boolean;
+  /**
+   * If autoResize, in here select the type, Width, height, or both
+   */
+  @Prop() autoResizeType: string;
+  /**
+   * Type of font
+   */
+  @Prop() fontFamily: string;
+  /**
+   * Font size
+   */
+  @Prop() fontSize: number;
+  /**
+   * Font Color
+   */
+  @Prop() fontColor: string;
+  /**
+   * Auto refresh group
+   */
+  @Prop() autoRefreshGroup: string;
+  /**
+   * Allowing or not Comlumn sort
+   */
+  @Prop() disableColumnSort: boolean;
+  /**
+   * Allow selection
+   */
+  @Prop() allowSelection: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XML
+   */
+  @Prop() exportToXML: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to HTML
+   */
+  @Prop() exportToHTML: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XLS
+   */
+  @Prop() exportToXLS: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XLSX
+   */
+  @Prop() exportToXLSX: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to PDF
+   */
+  @Prop() exportToPDF: boolean;
+  /**
+   * Title of the QueryViewer
+   */
+  @Prop() queryTitle: string;
+
+  private generateForm() {
+    const form = document.createElement("form");
+    form.action = this.baseUrl + this.mapServices[this.env];
+    form.method = "POST";
+    form.target = "query_viewer";
+    form.innerHTML = this.postData();
+
+    document.body.append(form);
```

Comment:
I think this could be done inside the `render` method, taking advantage of JSX, instead of manipulating the DOM.
Moreover, it would allow you to simplify all the logic that concatenates strings to create the form.

Reply:
I did it this way, because I needed to submit the form programmatically.

I tried to put in the render method, but I hadn't submit method that way.

Reply:
After rendering it in the `render` method with JSX, you can query the DOM for it using `const form = this.element.querySelector("form");`.

And then you can perform the submit programmatically.